test/pkg/environment/common/setup.go

```diff
@@ -96,7 +96,12 @@ func (env *Environment) ExpectCleanCluster() {
 		gvk := lo.Must(apiutil.GVKForObject(obj, env.Client.Scheme()))
 		metaList.SetGroupVersionKind(gvk)
 		Expect(env.Client.List(env.Context, metaList, client.Limit(1))).To(Succeed())
-		Expect(metaList.Items).To(HaveLen(0), fmt.Sprintf("expected no %s to exist", gvk.Kind))
+		if len(metaList.Items) > 0 {
+			for _, item := range metaList.Items {
+				// only allow system-surge and default node pools to exist for nap purposes
+				Expect(item.GetName()).To(Or(Equal("system-surge"), Equal("default")), fmt.Sprintf("expected only system-surge and general-purpose %s to exist", gvk.Kind))
```

Comment:
If its possible to pass in a flag, I'd a way to indicate we are using NAP, or self-hosted. 

It would allow us to have any differences clear within the code, as having clear points like this were we can be like `if NAP then ...` would allow us, and others to easily track any differences.